server.go

```diff
@@ -0,0 +1,75 @@
+package unixtransport
+
+import (
+	"fmt"
+	"net"
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"path/filepath"
+)
+
+// A TestServer is an HTTP+UNIX server listening on a temporary UNIX socket, for
+// use in end-to-end HTTP tests. TestServer's API mimics that of the
+// net/http/httptest.Server type.
+type TestServer struct {
+	// Base URL of form http+unix:///tmp/(random)/test.sock: with no trailing
+	// slash.
+	URL string
+
+	server *httptest.Server
+	client *http.Client
+
+	// Temporary directory, cleaned up on close.
+	dir string
+}
+
+// NewTestServer starts and returns a new TestServer. The caller should call
+// Close when finished, to shut it down.
+func NewTestServer(handler http.Handler) *TestServer {
+	// If any subsequent calls fail before we return *TestServer, be sure to
+	// os.RemoveAll(dir) to tidy things up.
+	dir, err := os.MkdirTemp(os.TempDir(), "unixtransport-server")
+	if err != nil {
+		panicf("failed to make temporary directory: %v", err)
```

Comment:
...this to `tb.Fatal` instead of panicking?

Reply:
Ack.